eth-voting-vue/contracts/SimpleVote.sol

```diff
@@ -4,24 +4,24 @@ contract SimpleVote {
     
     // simply iterate over arrays, with totalVotes - 1 as an index
     address[] public voters;
-    int8[] public votes;
+    //int8[] public votes;
     int256 public totalVotes;
     int8 public maxVoteValue;
     
     mapping (address=> int8) public getVote;
+    mapping (address=> bool) public hasVoted;
 
     constructor (int8 _maxVoteValue) public {
         maxVoteValue = _maxVoteValue;
     }
     
     function vote(int8 _vote) public returns (bool) {
-        // TODO: Uncomment next line in production, testing is easier like this now. It's prohibiting users from votig multiple times.
-        //require(getVote[msg.sender] == 0, "Sender has voted already");
         require(_vote <= maxVoteValue, "Voted for value higher than allowed");
         getVote[msg.sender] = _vote;
-        voters.push(msg.sender);
-        votes.push(_vote);
-        totalVotes++;
+        if(hasVoted[msg.sender] == false) {
+            hasVoted[msg.sender] == true
+            voters.push(msg.sender);
```

Comment:
why is `voters` and `totalVotes` needed?

Reply:
good point, replaced with getter function reading the voters array length.